mantidproject/mantid
# Allow log bins for QXY

**Description of work.**
This PR allows passing of logarithmic bins for 2D reduction in the SANS v2 GUI.

**Report to:** nobody

**To test:**
1. Interfaces -> SANS -> SANS v2 experimental
2. Load the attached user file. This user file has a line `L/ QXY 0.005 1.0 0.05 /log`.
3. Go to settings (lhs) and Q, Wavelength, Detector limits
4. Check that in Q limits QXY has a max 1.0, "step" (dQ/Q for log bins) of 0.05 and the combo box is set to log
5. Change the combo box to lin and check that only the Qxy "step" line edit changes title
6. Change the combo box back to log and go to the run tab
7. Load the attached batch file and set the reduction mode to 2D
8. Click process all. Check that there are no errors reducing the data.

Fixes #24878 

**Data**
[QXY_PR.zip](https://github.com/mantidproject/mantid/files/2900600/QXY_PR.zip)


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
